app/lib/foreman_deployments/tasks/wait_until_built_task_definition.rb

```diff
@@ -0,0 +1,60 @@
+module ForemanDeployments
+  module Tasks
+    class WaitUntilBuiltTaskDefinition < ForemanDeployments::Tasks::BaseDefinition
+      class Action < BaseAction
+        include Dynflow::Action::Polling
+
+        def done?
+          external_task['built'] == true
+        end
+
+        def timeout
+          input['timeout'] || 2 * 60 * 60 # 2 hours default
+        end
+
+        def invoke_external_task
+          schedule_timeout(timeout) unless timeout <= 0
+          WaitUntilBuiltTaskDefinition.build_status
+        end
+
+        def poll_external_task
+          # TODO: check the case when no host_id is sent
+          host = Host.find(input['host_id'])
+          WaitUntilBuiltTaskDefinition.create_output(host, output)
+          WaitUntilBuiltTaskDefinition.build_status(host)
+        end
+
+        def poll_interval
+          30
+        end
+      end
+
+      def validate
+        ValidationResult.new
+      end
+
+      def preliminary_output(_parameters)
+        WaitUntilBuiltTaskDefinition.create_output(Host::Managed.new)
```

Comment:
Probably a bad one: try to find a host which is already built. In a mature system it's not such a bad idea, but for empty systems it will become a problem. Additionally it adds kind of non-deterministic behavior to the system.
Let's leave this as is for now, and think about it later.